File: app/src/main/java/data/datamodels/Sources.java
package data.datamodels;


import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Entity(tableName = "sources",
indices = {@Index("id")})

public class Sources {

    @SerializedName("id")
    @Expose
    @NonNull
    @PrimaryKey
    private String id;

    @SerializedName("name")
    @Expose
    private String name;

    @SerializedName("description")
    @Expose
    private String description;

    @SerializedName("url")
    @Expose
    private String url;

    @SerializedName("category")
    @Expose
    private String category;


    @SerializedName("language")
    @Expose
    private String language;


    @SerializedName("country")
    @Expose
    private String country;



    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public String getLanguage() {
        return language;
    }

    public void setLanguage(String language) {
        this.language = language;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUrl() {
        return url;
    }
    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

}

File: app/src/main/java/com/example/newsapp/data/datamodels/Sources.kt
package com.example.newsapp.data.datamodels


import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

@Entity(tableName = "sources", indices = [Index("id")])
class Sources {

    @SerializedName("id")
    @Expose
    @PrimaryKey
    lateinit var id: String

    @SerializedName("name")
    @Expose
    var name: String? = null

    @SerializedName("description")
    @Expose
    var description: String? = null

    @SerializedName("url")
    @Expose
    var url: String? = null

    @SerializedName("category")
    @Expose
    var category: String? = null


    @SerializedName("language")
    @Expose
    var language: String? = null


    @SerializedName("country")
    @Expose
    var country: String? = null

}


